sys-utils/dmesg.c

```diff
@@ -1344,10 +1344,12 @@ static void print_record(struct dmesg_control *ctl,
 
 done:
 	free(mesg_copy);
-	if (ctl->json)
+	if (ctl->json) {
 		ul_jsonwrt_object_close(&ctl->jfmt);
-	else
+		ul_jsonwrt_flush(&ctl->jfmt);
```

Comment:
What about to add `if (ctl->follow)` before the flash? It will reduce overhead for normal usually use cases.

Reply:
Done